src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs

```diff
@@ -164,6 +163,10 @@ await ApplyChangeToWorkspaceAsync(w =>
                 onAfterUpdate: null);
         }).ConfigureAwait(false);
 
+        // Set this value early after solution is created so it is available to Razor.  This will get updated
```

Comment:
Not sure why Razor is being commented here -- this is just doing the thing the API is asking us to do....

Reply:
the early set was initially added for Razor here - https://github.com/dotnet/roslyn/commit/863962d8dd2e0151b1387259eeef0cbf66dfba75

It does get set anyway later on when the cmdline is set.